relay-server/src/metrics_extraction/transactions.rs

```diff
@@ -152,10 +168,73 @@ fn extract_user_satisfaction(
     None
 }
 
+/// These are the tags that are added to all extracted metrics.
+#[cfg(feature = "processing")]
+fn extract_universal_tags(
+    event: &Event,
+    custom_tags: &BTreeSet<String>,
+) -> BTreeMap<String, String> {
+    let mut tags = BTreeMap::new();
+    if let Some(release) = event.release.as_str() {
+        tags.insert("release".to_owned(), release.to_owned());
+    }
+    if let Some(dist) = extract_dist(event) {
+        tags.insert("dist".to_owned(), dist);
+    }
+    if let Some(environment) = event.environment.as_str() {
+        tags.insert("environment".to_owned(), environment.to_owned());
+    }
+    if let Some(transaction) = event.transaction.as_str() {
+        tags.insert("transaction".to_owned(), transaction.to_owned());
+    }
+
+    // The platform tag should not increase dimensionality in most cases, because most
+    // transactions are specific to one platform
+    let platform = event.platform.as_str().unwrap_or("other");
+    let platform = if store::is_valid_platform(platform) {
+        platform
+    } else {
+        "other"
+    };
```

Comment:
Does rust allow to do something like this?
```suggestion
    if !store::is_valid_platform(platform) {
        let platform = "other"
    };
```
If `platform` is valid, no need to shadow it.

Reply:
The second `let` would create an unused local variable inside the if branch, but you could make platform mutable from the start. But the most rusty way I can think of is this:
```rust
let platform = match event.platform.as_str() {
    Some(platform) if store::is_valid_platform(platform) => platform,
    _ => "other",
};
```